src/develop/imageop.c

```diff
@@ -2279,6 +2268,16 @@ void dt_iop_request_focus(dt_iop_module_t *module)
      && darktable.view_manager->accels_window.sticky)
     dt_view_accels_refresh(darktable.view_manager);
 
+  if((out_focus_module && out_focus_module->operation_tags_filter())
+     || (module && module->operation_tags_filter()))
+  {
+    dt_dev_invalidate_all(dev);
+    dt_dev_pixelpipe_rebuild(dev);
+    // don't use previous image as overlay
+    if(dev->pipe) dev->pipe->backbuf_zoom_x = 1000;
+    if(dev->preview2_pipe) dev->preview2_pipe->backbuf_zoom_x = 1000;
+  }
```

Comment:
I didn't see this before. We could use `dt_dev_pixelpipe_cache_invalidate_later()` here to keep modules caches of earlier modules in the pipeline available.